deluge/ui/sessionproxy.py

```diff
@@ -145,7 +145,13 @@ def get_torrent_status(self, torrent_id, keys):
 
                 def on_status(result, torrent_id):
                     t = time()
-                    self.torrents[torrent_id][0] = t
+                    torrent = self.torrents.get(torrent_id)
+                    if not torrent:
+                        log.debug(
+                            f'Got status for unidentified torrent id {torrent_id}'
+                        )
+                        return {}
+                    torrent[0] = t
                     self.torrents[torrent_id][1].update(result)
```

Comment:
I think you could run into the same problem here if the torrent is removed after you get it from the self.torrents dict. Maybe doing the same thing we do below in get_torrent_statuses() (wrap the body of this function in a try/except on KeyError) would be the better approach?

Reply:
It might be a better approach. Pushed a fix.